### PR TITLE
Fix save errors for currency.json

### PR DESCRIPTION
**Pull request by NotJustSid**
- `Module::Save(...)` now also truncates the file when opening it.
- Moved some user removal (for users who left the server) code from `CurrencyModule::OnSave(...)`

### CODE REVIEW

File: src/modules/CurrencyModule.cpp [hunk 1]
```diff
@@ -18,28 +18,35 @@
...
-					user->isDailyClaimed = false;
+					user.isDailyClaimed = false;
 				}
-
+				else 
+				{
+					//remove if the user is no longer in the server
+					guildList[server].removeAt(i);
+				}
+			}
```
Comment (Gaztin) [inline src/modules/CurrencyModule.cpp]: What happened to the indentation here?

**Review by Gaztin — CHANGES_REQUESTED** [review]

File: src/modules/CurrencyModule.cpp [hunk 2]
```diff
@@ -985,11 +991,8 @@ void CurrencyModule::OnSave(QJsonDocument& doc) const
 	{
 		QJsonObject serverJSON;
 		
-		for (auto user = guildList[server].begin(); user != guildList[server].end(); user++) {
-			if (m_client->GetName(server, user->userId) == "")
-			{
-				continue;
-			}
```
Comment (Gaztin) [inline src/modules/CurrencyModule.cpp:992]: Why do I get a feeling that this was the bad guy all along..

File: src/modules/CurrencyModule.cpp (hunk 1 above)
Comment (NotJustSid) [reply src/modules/CurrencyModule.cpp]: The indentation is all fine?

Comment (NotJustSid) [reply src/modules/CurrencyModule.cpp]: Ok it's fine in the code highlighted here, but it's not in the section that follows (EDIT: and the one just before)... On it.

File: src/modules/CurrencyModule.cpp (hunk 2 above)
Comment (NotJustSid) [reply src/modules/CurrencyModule.cpp:992]: It's what *does* the bad job, but it isn't what causes the issue...

**Review by Gaztin — APPROVED** [review]

**Pull request merged**